### DISCUSSION TITLE
Design improvement discussion - Episodes

### DISCUSSION THREAD

**Discussion by dzuk-mutant** [datafruits/datafruits#1298] [Ideas]
_This thread is continuing design conversations had in the Discord. These are broken out into different areas and parts of the site to enable more structured discussion and also so these items can be worked on atomically with the aim of eventually improving the overall frontend design._

This thread covers episodes - **both episode card views that are seen in the archive, show pages, etc. as well as episode pages that show just one episode**. I felt like combining both would enable us to take design continuity into consideration, so no matter how an episode is viewed across the website, any representation of an episode has some similarities and they feel like part of a broader whole.

**Comment by dzuk-mutant** [comment]
I'm re-iterating my thoughts and ideas about episodes from Discord conversations here:

## Get DJs to deduplicate metadata for future shows

Currently, air date information about an episode is primarily represented in the title and it is in a MMDDYYYY format. There's arguably a few problems with this practice:

- Short dates without separators are hard to read
- It's redundant information because there's already separate metadata for air dates in the episodes.
- It's in a format that only Americans use, which can create confusion.

Taking this date out of the show metadata would enable layouts about episodes to flow much more smoothly, and also let the website show the air date to the user in a format that they are more likely to readily understand, something that Air Dates in episode pages already do.

Episode titles also contain the host of the show whether or not they are a guest. This is also a duplication of information as the host of the show is already in the show metadata and it prevents layouts from being smoother and easier to read.

So my suggestion would be to ask DJs to:

- Do not put dates in show titles
- Do not put the main host(s) in the show titles. Feel free to put a DJ name there if there is a guest mix.

This will result in shows before this change appearing a bit redundant, but I think it'll still look better overall and it will also just lead to more readable layouts (where dates can be localised to the viewer) and more resilient metadata structures.

This will also require changes to the documentation and letting all current DJs know how to do show metadata in the future. 

## /podcasts and episode card views
I already proposed some changes with [this PR](https://github.com/datafruits/datafruits/pull/1285), this list adds to those:
- Show a short air date that's automatically localised to the viewer (eg. MM/DD/YYYY, YYYY-MM-DD, etc.)
- Clearly differentiate between broadcasts that have recordings and those that don't.
- Fix whatever issue is preventing podcast episodes from downloading from the browser when the download button is clicked. (As mentioned in the Discord, it may be a CORS issue?)
- If possible, unify all episode card views to one view, where layout components can be added and taken away based on the needs of the page.

## Specific episode page
- Make the artwork container a square aspect ratio as DJs are encouraged to upload in this format.
- Make the artwork not overbearing in the layout.
- Show the Episode metadata first in sequence.
- Change the link that lets users see more of a show to 'check out more episodes of this show', to prevent potential repetition as the episode title is fairly likely to have the show title already.
- Replace the tag view here with the new unified tag view (which I started in [my PR](https://github.com/datafruits/datafruits/pull/1285))
- Remove duplicate tag views.
- Make the player/download/etc controls more prominent.
- Streamline the comment section styling so it flows and scans better.
- Clearly differentiate between episodes that have recordings and those that don't.
- Fix the aforementioned issue that is preventing podcast episodes from downloading from the browser when the download button is clicked.

----

## A few design ideas

And here are a couple of design ideas for episode pages that I came up with based on these thoughts and other ideas (also already seen on the Discord!);

![image](https://github.com/datafruits/datafruits/assets/32108484/1c0a7c2f-a997-4731-a7d7-a3eef0a5faa0)
![image](https://github.com/datafruits/datafruits/assets/32108484/9058b4d7-777c-4831-b7a1-aeac130a5a60)

**Comment by dzuk-mutant** [comment]
A few more dev-centric things I noticed while looking at the code:

- the `alt` of the episode artwork in an episode is just the show title. This should be a description of the episode art.
- The 'Check out more episodes of X' string is not a string for localisation (hardcoded English).

**Comment by dudethatbe** [comment]
Love the ideas here and focusing on the archives/shows like this. The only thing I had a hard time understanding was this point:
> If possible, unify all episode card views to one view, where layout components can be added and taken away based on the needs of the page.

Is this so we could continually load a stream of episodes (instead of hard-coded pagination) ?

**Reply by mcfiredrill** [reply]
> If possible, unify all episode card views to one view, where layout components can be added and taken away based on the needs of the page.

I assumed this was more about refactoring the card views to use the same component so its consistent across pages?

**Reply by dzuk-mutant** [reply]
Yeah what @mcfiredrill said is what I meant ^^', in some of my own code, I refactored the card views a bit so they're consistent but there's still others that are not (such as the episode views on a show page).

Keep the code DRY as well as making the appearance of the same types of data consistent across the site.
